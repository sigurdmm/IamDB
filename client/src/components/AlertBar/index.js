import React from 'react';
import PropTypes from 'prop-types';
import './index.less';

const AlertBar = ({
  type,
  message,
}) => {
  if (type === 'warning') {
    return <p className='alertbar__warning'>{message}</p>;
  }
  return <p className='alertbar__error'>{message}</p>;
};

AlertBar.propTypes = {
  message: PropTypes.string.isRequired,
  type: PropTypes.string,
};

export default AlertBar;
