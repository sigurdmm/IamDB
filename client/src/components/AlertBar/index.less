@import '../../constants';
.alertbar__error {
  background-color: @errorRed;
  color: @white;
  font-size: 20px;
  padding: 1em;
  overflow-wrap: break-word;
}
