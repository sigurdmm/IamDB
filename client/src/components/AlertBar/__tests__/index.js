import { shallow } from 'enzyme';

import AlertBar from '../index';


describe('<AlertBar/>', () => {
  it('should render properly', () => {
    const wrapper = shallow(<AlertBar
      message={''}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
