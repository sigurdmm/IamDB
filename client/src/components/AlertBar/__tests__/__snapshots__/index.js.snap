// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertBar/> should render properly 1`] = `
<p
  className="alertbar__error"
/>
`;
